### PR TITLE
[REVIEW] Added build.sh script, updated CI scripts and documentation

### PR DESCRIPTION
**Pull request by rlratzel**
This PR adds a new top-level script named `build.sh` that allows a contributor to build from source using the same commands and options the CI system, container builds, doc builds, nightly unit test runs, and (eventually) ASV build & test iterations use. The relevant CI scripts have been updated to call this script and all related documentation should be updated as well.

This PR allows a maintainer to update the common build steps in one place while also simplifying the build process for all contributors and other build-related operations.

The changes were tested using the gpuCI container via the `ci/local/build.sh` testing script, where cudf branch-0.8 was successfully built and tested without errors.

### CODE REVIEW

**Comment by dillon-cullinan** [comment]
I think the script locations should be "swapped."

The full source build code being in `gpu/build.sh` and you have `build.sh` in root pointing to that. Then set the default args for `gpu/build.sh` to work with CI.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -105,30 +105,21 @@ source activate cudf_dev
...
-
-# CMake options:
-# -DCMAKE_INSTALL_PREFIX set to the install path for your libraries or $CONDA_PREFIX if you're using Anaconda, i.e. -DCMAKE_INSTALL_PREFIX=/install/path or -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX
-# -DCMAKE_CXX11_ABI set to ON or OFF depending on the ABI version you want, defaults to ON. When turned ON, ABI compability for C++11 is used. When OFF, pre-C++11 ABI compability is used.
-$ cmake .. -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX -DCMAKE_CXX11_ABI=ON      # configure cmake ...
-
-$ make -j                                                                 # compile the libraries librmm.so, libcudf.so ... '-j' will start a parallel job using the number of physical cores available on your system
-$ make install                                                            # install the libraries librmm.so, libcudf.so to the CMAKE_INSTALL_PREFIX
+$ cd $CUDF_HOME
+$ ./build.sh libcudf                   # compile the cuDF libraries and install them to $INSTALL_PREFIX if set, otherwise $CONDA_PREFIX
```
Comment (kkraus14) [inline CONTRIBUTING.md]: I do not think we should hide the build steps underneath a shell script. We should offer this as an alternative for a "streamlined" process

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -105,30 +105,21 @@ source activate cudf_dev
...
 $ make test
 ```
 
-- Build the `cudf` python package, in the `python` folder:
+- Build the `cudf` python package and install to your Python path:
 ```bash
-$ cd $CUDF_HOME/python
-$ python setup.py build_ext --inplace
+$ cd $CUDF_HOME
+$ ./build.sh cudf
```
Comment (kkraus14) [inline CONTRIBUTING.md]: Same here, don't hide it.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -139,14 +130,22 @@ NUMBAPRO_LIBDEVICE=$CUDA_HOME/nvvm/libdevice
...
-$ py.test -v                                        # run python tests on cudf python bindings
+$ py.test -v                           # run python tests on cudf python bindings
 ```
 
-- Finally, install the Python package to your Python path:
+- Other `build.sh` options:
 ```bash
-$ python setup.py install                           # install cudf python bindings
+$ cd $CUDF_HOME
+$ ./build.sh clean                     # remove any prior build artifacts and configuration (start over)
```
Comment (kkraus14) [inline CONTRIBUTING.md]: ❤️

File: CONTRIBUTING.md (hunk 1 above)
Comment (rlratzel) [reply CONTRIBUTING.md]: Done. Let me know if it still needs improvement.

File: CONTRIBUTING.md (hunk 2 above)
Comment (rlratzel) [reply CONTRIBUTING.md]: Done. Let me know if it still needs improvement.

**Comment by rlratzel** [comment]
> I think the script locations should be "swapped."
> 
> The full source build code being in `gpu/build.sh` and you have `build.sh` in root pointing to that. Then set the default args for `gpu/build.sh` to work with CI.

Thanks @dillon-cullinan , I'm thinking CI is one of several consumers of the "common" build steps though, and not necessarily the owner. This approach prevents CI-isms from slipping in to the common build steps, which may otherwise break other non-CI use cases.

**Pull request merged**